src/BaseGL.js

```diff
@@ -321,10 +344,124 @@ class BaseGL {
+
+      // Only run this code if hi
+      if (hdata && hdata.indices.length > 0 && this.nrows) {
+        const index = hdata.indices[0]; // handle only one point
+        const col = Math.floor(index / this.nrows);
+        const row = index % this.nrows;
+
+        // Invert row, considering X axis starts from bottom up
+        const rowInverted = this.nrows - 1 - row;
+        hdata["row"] = rowInverted;
+        hdata["col"] = col;
+      }
+
+      if (this.highlightEnabled && hdata && hdata.indices.length > 0) {
+        this.highlightIndices(hdata.indices);
+      }
+
       self.clickCallback(hdata);
     });
   }
 
+  highlightIndices(indices, forceSet = false) {
+    if (forceSet) {
+      this.highlightedIndices = [...indices];
+    } else if (this.highlightedIndices.length > 0) {
+      indices.forEach((index) => {
+        const foundIndex = this.highlightedIndices.indexOf(index);
+        if (foundIndex > -1) {
+          this.highlightedIndices = this.highlightedIndices.filter(
+            (item) => item !== index
+          );
+        } else {
+          this.highlightedIndices.push(index);
+        }
+      });
+    } else {
+      this.highlightedIndices.push(...indices);
+    }
+    this.highlightedIndicesCallback(this.highlightedIndices);
+    this.reRender();
+  }
+
+  /**
+   * Enable highlight for the plot. This is useful when the plot is rendered with
+   * a subset of data and we want to highlight the points that are not rendered.
+   * @memberof BaseGL
+   * @example
+   * // Enable highlight
+   * plot.enableHighlight();
+   */
+  enableHighlight() {
+    this.highlightEnabled = true;
+  }
+
+  /**
+   * Disable highlight for the plot. This is useful when the plot is rendered with
+   * a subset of data and we want to highlight the points that are not rendered.
+   * @memberof BaseGL
+   * @example
+   * // Disable highlight
+   * plot.disableHighlight();
+   */
+  disableHighlight() {
+    this.highlightEnabled = false;
+    this.clearHighlight();
+  }
+
+  /**
+   * Clear the highlight for the plot.
+   * @memberof BaseGL
+   * @example
+   * // Clear highlight
+   * plot.clearHighlight();
+   **/
+  clearHighlight() {
+    this.highlightedIndices = [];
+    this.highlightedIndicesCallback(this.highlightedIndices);
+    this.reRender();
+  }
+
+  /**
+   * Re-render the plot. This is useful when the data is updated.
+   * @memberof BaseGL
+   */
+  reRender() {
+    this.plot.updateSpecification({
+      ...this._spec,
+      defaultData: {
+        ...this._spec.defaultData,
+        color: this._spec.defaultData.color.map((color, index) => {
+          if (
+            this.highlightedIndices.length === 0 ||
+            this.highlightedIndices.includes(index)
+          ) {
+            return color;
+          } else {
+            const rgb = isRGB(color) ? strToRGB(color) : hexToRGB(color);
+            const dimmedRgb = mixWithWhite(rgb, 0.7); // 0.7 is the dimming factor
```

Comment:
I assume this is due to opacity not working as expected? Have you been able to determine where opacity is handled differently than color? I imagine this might be related due to the data not being a one-dimensional (array-like), but rather two-dimensional (matrix-like).

Reply:
Yes, True. Since opacity wasn't working I dimmed the colors manually by playing with rgb color scheme.

I'll look into it if I can use opacity array instead. Thanks.

Reply:
Sounds good. I haven't seen how the color dimming looks like (you can demo tomorrow maybe?), but that may be a good alternative if opacity is tricky.

Reply:
Sure. We can do it today or tomorrow.

Reply:
> opacity not working as expected

can you help me understand this? trying to figure out where the issue is

Reply:
> > opacity not working as expected
> 
> can you help me understand this? trying to figure out where the issue is

I wanted to change the opacity of the cells on highlight operation. Selected items as fully opaque and unselected items a bit transparent. So I created a 1d array similar to colors and passed them different values. But no cell was being rendered as I expected them to be based on opacity array data. I don't know if that opacity array is for this purpose or am I doing something wrong.

So for now I mixed the rgb with white to make it work.

Reply:
I guess, the appropriate way of doing this is to set the alpha when we unpack color and pass in the opacity here - https://github.com/epiviz/epiviz.gl/blob/main/src/epiviz.gl/webgl.js#L28

Reply:
@jkanche, Thanks figured it out. Updated the code.